mlrun/runtimes/nuclio/application/application.py

```diff
@@ -269,6 +292,9 @@ def deploy(
             builder_env,
         )
 
+        ports = self.spec.internal_application_port if direct_port_access else []
+        self.create_api_gateway(skip_access_key_auth=skip_access_key_auth, ports=ports)
+
```

Comment:
More a question than a comment. Do we want to provide a way to create an api gateway with skip_access_key only? What if user wants to create it with BasicAuth?

Reply:
That a valid question.
Users can still manually create api gateways with different configuration that will direct to the application runtime.
The question is what should be the default behavior.
Thinking about it, we might want to accept the `mode` as an argument, and perhaps have some default mode in the mlrun config.

Reply:
@TomerShor as an option, we can do the same way we do in api gateway init. Which means passing `APIGatewayAuthenticator` when create api gateway. But in case of application runtime we can create an api gateway with `accessKey` by default.

Reply:
I prefer using a `mode` to allow changing defaults in mlrun config, and use the `with_XXX` methods for the API Gateway, instead of forcing the user to pass a `APIGatewayAuthenticator` object (a bit more complex usage IMO).